### DISCUSSION TITLE
Lets Start the Website

### DISCUSSION THREAD

**Discussion by vybhav72954** [HarshCasper/Rotten-Scripts#748] [Ideas]
@HarshCasper I am proposing we migrate #584 here.

We can get started with the website from the weekend.

Considering Rotten-Script has grown enough, it's time we start with the Website.

**Proposal**

- BACKEND

Node.js with Express.js
Mongo DB with Mongoose

- FRONTEND

React.js

- SERVER SIDE

Any Plugin with React.js

Required -  
Admin Panel
Folder Structure just like the Repo
Using README for description

If you allow, I would like to make a separate Repo, (cause Branching might work for static website), but I doubt we should have the whole CRUD framework on this Repo itself.

**Comment by avinal** [comment]
Hello @vybhav72954, You might have missed some discussions, I am sorry that I was not able to update the discussions as directed by @HarshCasper. I will add the progress by today. In short, a static site based on the sphinx book theme is already in progress. You might see the initial pages at https://avinal.space/Rotten-Scripts, there are some problems though but will be corrected with time. I don't know if there is a second website project too. Harsh will confirm that. Thanks

**Reply by vybhav72954** [reply]
Hey man it looks so good 😮 

I am sorry I was not aware of that that's why I started this discussion over here.
This is simple and awesome. @HarshCasper I think @avinal is handling this really well.

Let's hope that we can deploy this asap to increase the reach 🎆

**Reply by avinal** [reply]
Hehe, no worries @vybhav72954, I hope it will be deployable in a day or two. As I said in the other discussion #755, it is barely a skeleton, there will be a lot of work to improve pages. You are welcome to contribute 🤗.

**Reply by vybhav72954** [reply]
Trust me if I was any good at a static website, It would have been my pleasure, i never ever gave it a shot 😓

I might try to learn something about it though 🤩

**Reply by avinal** [reply]
@vybhav72954 I think you can integrate SEO and other web-based extensions to the website. I am not a web developer 😅so don't know much about them.

**Comment by HarshCasper** [comment]
Hi @vybhav72954 

Thanks for kick-starting this! 

@avinal is quite right on the same. He has been working hard on the Static Website which will be ready for deployment pretty soon, I guess. 

Now the second question comes, do we actually need a CRUD App? I would like to know the merits and demerits for the same, and to understand how a CRUD App will help us better than a Static Website. 

Would love to know your opinions on the same @vybhav72954 @avinal 

Tagging some previous maintainers @seema1711 and @srimani-programmer to see if they have any view points on the same

**Reply by iseemasaharan** [reply]
https://avinal.space/Rotten-Scripts
This looks simple and pretty. 🙌

**Reply by vybhav72954** [reply]
@HarshCasper The reason I believe having a dynamic website is better is because of the user interactivity and in turn the reach it generates for us.

But considering that the Static website project is already on, we can shelf this.

The thing is i was planning to restructure Repo a bit, CLassify the scripts in say sub directories For example, script that incorporates Scrappin technique, OCR techniques, ML DL techniques etc. 

In that way, the website would become more of a curated list, cause i think that doing that with Repo will make it only more tougher for the new coming contributors.

But like i said, if the static website is already ready, we can focus on that

**Reply by HarshCasper** [reply]
That's a good point. 

Maybe we can start a new discussion and boil out new ideas that we can work on. As far as I can notice for the upcoming few months, we can focus on these: 

- Make the repo cleaner
- Having the Static Site up and ready
- Classify the Scripts in directories
- Have some utility scripts to help the Project overall
- Making the Contribution experience better (With Github Actions)
- Plan out deliverables for a Dynamic Site
